qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -1063,6 +1063,158 @@ def GaussianSquareDrag(
+) -> SymbolicPulse:
+    """An echoed Gaussian square pulse with an active tone overlaid on it. 
+    Exactly one of the ``risefall_sigma_ratio`` and ``width`` parameters has to be specified.
+    If ``risefall_sigma_ratio`` is not ``None`` and ``width`` is ``None``:
+    .. math::
+        \\text{risefall} &= \\text{risefall_sigma_ratio} \\times \\text{sigma}\\\\
+        \\text{width} &= \\text{duration} - 2 \\times \\text{risefall}
+    If ``width`` is not None and ``risefall_sigma_ratio`` is None:
+    .. math:: \\text{risefall} = \\frac{\\text{duration} - \\text{width}}{2}
+    Gaussian :math:`g(x, c, σ)` and lifted gaussian :math:`g'(x, c, σ)` curves
+    can be written as:
+    .. math::
+        g(x, c, σ) &= \\exp\\Bigl(-\\frac12 \\frac{(x - c)^2}{σ^2}\\Bigr)\\\\
+        g'(x, c, σ) &= \\frac{g(x, c, σ)-g(-1, c, σ)}{1-g(-1, c, σ)}
+    
+    Args:
+        duration: Pulse length in terms of the sampling period `dt`.
+        amp: The amplitude of the rise and fall and of the square pulse.
+        sigma: A measure of how wide or narrow the risefall is; see the class
+               docstring for more details.
+        width: The duration of the embedded square pulse.
+        angle: The angle in radians of the complex phase factor uniformly
+            scaling the pulse. Default value 0.
+        active_amp: The amplitude of the active cancellation tone.
+        active_angle: The angle in radian of the complex phase factor uniformly
+            scaling the active pulse. Default value 0.
+        risefall_sigma_ratio: The ratio of each risefall duration to sigma.
+        name: Display name for this pulse envelope.
+        limit_amplitude: If ``True``, then limit the amplitude of the
+            waveform to 1. The default is ``True`` and the amplitude is constrained to 1.
+    Returns:
+        ScalableSymbolicPulse instance.
+    Raises:
+        PulseError: When width and risefall_sigma_ratio are both empty or both non-empty.
+    """
+    # Convert risefall_sigma_ratio into width which is defined in OpenPulse spec
+    if width is None and risefall_sigma_ratio is None:
+        raise PulseError(
+            "Either the pulse width or the risefall_sigma_ratio parameter must be specified."
+        )
+    if width is not None and risefall_sigma_ratio is not None:
+        raise PulseError(
+            "Either the pulse width or the risefall_sigma_ratio parameter can be specified"
+            " but not both."
+        )
+
+    if width is not None and risefall_sigma_ratio is None:
+        total_risefall = duration - width
+        echo_risefall = 2*total_risefall
+        width_echo = (duration - echo_risefall)/2
+
+    if width is None and risefall_sigma_ratio is not None:
+        width = duration - 2.0 * risefall_sigma_ratio * sigma
+        width_echo = (duration - 4.0 * risefall_sigma_ratio * sigma)/2
+
+
+
+    parameters = {"amp": amp, "angle": angle, "sigma": sigma, "width": width, "width_echo": width_echo, 
```

Comment:
`width_echo` should be `width/2`, so we don't need a separate pulse parameter for it.

Reply:
In this case I have been denoting as `width` the square part of the active cancellation tone (the one that lasts for the entire duration of the pulse) which has one rise and one fall part. `width_echo` is the width of the echo pulses, these have one rise and one fall part each (so two rises and two falls, all happening within the total duration) and thus their width is actually shorter than `width/2`. This is what `width_echo` captures.

Reply:
That is a fair point about it not actually being `width/2` but `width_echo` can still be inferred from the other pulses. `parameters` as passed to `SymbolicPulse` should be a minimal set of parameters to define the pulse. When the pulse is submitted to the backend, it is given as a pulse name and a dictionary of parameter names and values. I don't think it makes sense to have an extra parameter, just like how this function does not take `echo_width` as an extra parameter.

Reply:
Yeah, agreed